src/pages/ValidateLoginPage/index.website.js

```diff
@@ -117,20 +117,21 @@ class ValidateLoginPage extends Component {
     }
 
     render() {
-        const isTfaRequired = lodashGet(this.props, 'account.requiresTwoFactorAuth', false);
+        const is2FARequired = lodashGet(this.props, 'account.requiresTwoFactorAuth', false);
         const isSignedIn = Boolean(lodashGet(this.props, 'session.authToken', null));
+        const currentAuthState = this.getAutoAuthState();
         return (
             <>
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.FAILED && <ExpiredValidateCodeModal />}
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && (!isTfaRequired || isSignedIn) && <AbracadabraModal />}
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && isTfaRequired && !isSignedIn && <TfaRequiredModal />}
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.NOT_STARTED && (
+                {currentAuthState === CONST.AUTO_AUTH_STATE.FAILED && <ExpiredValidateCodeModal />}
+                {currentAuthState === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && is2FARequired && !isSignedIn && <TfaRequiredModal />}
+                {currentAuthState === CONST.AUTO_AUTH_STATE.NOT_STARTED && isSignedIn && <AbracadabraModal />}
+                {currentAuthState === CONST.AUTO_AUTH_STATE.NOT_STARTED && !isSignedIn && (
```

Comment:
@s77rt  Yes, it came from this [commit](https://github.com/Expensify/App/pull/19260/commits/4da2f319f04950908af5a04c434aba1510043252) to fix this bug mentioned in this [slack thread](https://expensify.slack.com/archives/C049HHMV9SM/p1684405641601719).

Before, it was necessary because the check for `AbracadabraModal` (now `JustSignedInModal`) used `NOT_STARTED` state (this is because the state was cleaned on startup - mentioned in the [proposal](https://github.com/Expensify/App/issues/19187#issuecomment-1554254779)), so we needed to differentiate between that one and the one for `ValidateCodeModal`. At the time, the only difference was that for the abracadabra, the `isSignedIn` was true and false for the `ValidateCodeModal`.

Reply:
@BeeMargarida Thank you!